pkg/p2p/peer/peer.go

```diff
@@ -245,10 +247,33 @@ func (p *Reader) ReadLoop() {
 			return
 		}
 
-		p.router.Collect(bytes.NewBuffer(b))
+		message, err := verifyChecksum(b)
+		if err != nil {
+			log.WithFields(log.Fields{
+				"process": "peer",
+				"error":   err,
+			}).Warnln("error reading message")
+			return
+		}
+
+		p.router.Collect(bytes.NewBuffer(message))
 	}
 }
 
+func verifyChecksum(m []byte) ([]byte, error) {
```

Comment:
`verifyChecksum` should be moved into a `checksum` package and renamed into `checksum.Extract`. This is consistent with `topics.Extract` and the fact that it mutates the input (a function named `verifyXXX` should not mutate input parameters and should simply return a `bool` or an `error`)